Userland/Utilities/tr.cpp

```diff
@@ -7,8 +7,155 @@
 #include <AK/Optional.h>
 #include <AK/String.h>
 #include <LibCore/ArgsParser.h>
+#include <ctype.h>
 #include <stdio.h>
 
+static const StringView s_alnum = "[:alnum:]";
+static const StringView s_blank = "[:blank:]";
+static const StringView s_digit = "[:digit:]";
+static const StringView s_lower = "[:lower:]";
+static const StringView s_punct = "[:punct:]";
+static const StringView s_upper = "[:upper:]";
+static const StringView s_alpha = "[:alpha:]";
+static const StringView s_cntrl = "[:cntrl:]";
+static const StringView s_graph = "[:graph:]";
+static const StringView s_print = "[:print:]";
+static const StringView s_space = "[:space:]";
+static const StringView s_xdigit = "[:xdigit:]";
+
+static bool is_octal(char c)
+{
+    return c == '0' || c == '1' || c == '2' || c == '3' || c == '4' || c == '5' || c == '6' || c == '7';
+}
+
+static int consume(StringView in, StringBuilder& out)
```

Comment:
This whole thing would be significantly more compact and readable if it used a `GenericLexer`, here's an outline:
```c++
static void consume(GenericLexer& lexer, StringBuilder& out)
{
    if (lexer.consume_specific("[:alnum:]"sv)) {
        for (int i = 0; i < 128; i++) {
            if (isalnum(i)) out.append(static_cast<char>(i));
        }
        return;
    }
    if (lexer.consume_specific("[:blank:]"sv)) {
        for (int i = 0; i < 128; i++) {
            if (isblank(i)) out.append(static_cast<char>(i));
        }
        return;
    }
    ...
    auto ch = lexer.consume_escaped_character();
    if (!lexer.consume_specific('-')) {
		out.append(ch);
        return;
	}
    auto range_end_ch = lexer.consume_escaped_character();
    for (auto c = ch; c < range_end_ch; ++c)
    	out.append(c);
}
```

This could be further simplified by just adding a local lambda that does the "iterate over 1-128 and append the matching characters" task, given the function.